bouncer/src/repo/manipulator/modelutility/repo_scene_manager.cpp

```diff
@@ -515,9 +515,29 @@ bool SceneManager::shouldGenerateSrcFiles(
 	const repo::core::model::RepoScene* scene,
 	repo::core::handler::AbstractDatabaseHandler* handler) const
 {
-	repo::core::model::RepoUser user(handler->findOneByCriteria(REPO_ADMIN, REPO_SYSTEM_USERS, BSON("user" << scene->getDatabaseName())));
 	//only generate SRC files if the user has the src flag enabled and there are meshes
-	return scene->getAllMeshes(repo::core::model::RepoScene::GraphType::DEFAULT).size() && user.isSrcEnabled();
+	
+	bool hasTriangleMeshes = false;
+	auto meshes = scene->getAllMeshes(repo::core::model::RepoScene::GraphType::DEFAULT);
+	for (const repo::core::model::RepoNode* node : meshes)
+	{
+		auto mesh = dynamic_cast<const repo::core::model::MeshNode*>(node);
+		if (!mesh)
+		{
+			continue;
+		}
+
+		if (mesh->getPrimitive() == repo::core::model::MeshNode::Primitive::TRIANGLES)
+		{
+			hasTriangleMeshes = true;
+			break;
+		}
+	}
+
+	repo::core::model::RepoUser user(handler->findOneByCriteria(REPO_ADMIN, REPO_SYSTEM_USERS, BSON("user" << scene->getDatabaseName())));
+	bool userHasSrcFlag = user.isSrcEnabled();
```

Comment:
@sebjf can we check for this first?

If the user doesn't have the SRC flag then we can return early and would not have to probe the meshes.

Reply:
@carmenfan Updated!